pages/leaderboard/index.tsx

```diff
@@ -11,16 +11,31 @@ import TableRow from "@mui/material/TableRow";
 import Paper from "@mui/material/Paper";
 import Header from "../../components/Header";
 import NavBar from "../../components/NavBar";
-import { participants } from "data/leaderboardUtils";
+import { useRouter } from "next/dist/client/router";
+import useSWR from "swr";
+import { Participant } from "models/leaderboardSchema";
+
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
 
 const LeaderboardPage: NextPage = () => {
+  const { data: participants, error } = useSWR<Participant[], unknown>(
+    "/api/leaderboardAPI/leaderboardMain",
+    fetcher,
+    {
+      refreshInterval: 5000,
+    }
+  );
+  const router = useRouter();
+  const navigateTo = function (to: string) {
+    router.push(to);
+  };
```

Comment:
This function is kinda unnecessary, right? It is just a wrapper for `router.push()` with the same signature. So at best it allows you to write one less word? 😬

Reply:
Yeah thought the same but the tutorial did it so thought it had some purpose :sweat_smile: Thank you! Removed it for now as it's not used and lint is complaining